multi/multi.go

```diff
@@ -482,7 +487,26 @@ func (connMulti *ConnectionMulti) EvalAsync(expr string, args interface{}) *tara
 	return connMulti.getCurrentConnection().EvalAsync(expr, args)
 }
 
+// ExecuteAsync passes sql expression to Tarantool for execution.
+func (connMulti *ConnectionMulti) ExecuteAsync(expr string, args interface{}) *tarantool.Future {
+	return connMulti.getCurrentConnection().ExecuteAsync(expr, args)
+}
+
+// NewPrepared passes a sql statement to Tarantool for preparation synchronously.
+func (connMulti *ConnectionMulti) NewPrepared(expr string) (*tarantool.Prepared, error) {
+	return connMulti.getCurrentConnection().NewPrepared(expr)
+}
+
 // Do sends the request and returns a future.
 func (connMulti *ConnectionMulti) Do(req tarantool.Request) *tarantool.Future {
+	if stickyReq, ok := req.(tarantool.ConnectedObject); ok {
+		_, belongs := connMulti.getConnectionFromPool(stickyReq.Conn().Addr())
+		if !belongs {
+			fut := &tarantool.Future{}
+			fut.SetError(tarantool.ErrStrangerConn)
+			return fut
```

Comment:
A single way to create an object simplifies life in the future:

NewFuture() + fut.SetError()
[2]

Reply:
Fixed.